### PR TITLE
Add Signup event.

### PR DESCRIPTION
**Pull request by pixlwave**
Adds an event to be sent when a new user accepts the analytics prompt if it was presented as part of the registration flow.

### CODE REVIEW

File: schemas/Signup.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      "oneOf": [
+        {"const": "Password", "description": "Registration with a username and password."},
+        {"const": "SignInWithApple", "description": "Social login using Apple."},
+        {"const": "SignInWithFacebook", "description": "Social login using Facebook."},
+        {"const": "SignInWithGitHub", "description": "Social login using GitHub."},
+        {"const": "SignInWithGitLab", "description": "Social login using GitLab."},
+        {"const": "SignInWithGoogle", "description": "Social login using Google."},
+        {"const": "SSO", "description": "Registration using another SSO provider."}
+      ],
+      "required": true
```
Comment (t3chguy) [inline schemas/Signup.json:20]: given this is required, there should be an `Other` for if fallback/etc auth is used

File: schemas/Signup.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    }
+  },
+  "required": ["eventName"],
```
Comment (t3chguy) [inline schemas/Signup.json:23]: also shouldn't the required attribute be specified here rather than up there?

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

File: schemas/Signup.json (hunk 2 above)
Comment (pixlwave) [reply schemas/Signup.json]: Yes good point. I did notice this when I opened the PR, but saw it done both ways in other events. That said updating it I noticed that quicktype doesn't honour it for TS so we should probably fix any `"required": true` properties.

Comment (pixlwave) [reply schemas/Signup.json]: Done ✅

File: schemas/Signup.json (hunk 1 above)
Comment (pixlwave) [reply schemas/Signup.json]: Makes sense, added 👍

File: schemas/Signup.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  "properties": {
+    "eventName": {
+      "enum": ["Signup"]
+    },
+    "authenticationType": {
+      "type": "string",
+      "description": "The type of authentication that was used to sign up.",
+      "oneOf": [
+        {"const": "Password", "description": "Registration with a username and password."},
+        {"const": "SignInWithApple", "description": "Social login using Apple."},
```
[the remark this answers is not in the document]
Comment (pixlwave) [reply schemas/Signup.json:13]: Happy to remove 👍

**Review by ouchadam — APPROVED** [review]

**Pull request merged**